zilliqa/src/state.rs

```diff
@@ -158,6 +156,17 @@ impl State {
         })
     }
 
+    pub fn mutate_account<F: FnOnce(&mut Account)>(
```

Comment:
This is a great idea. How easily can we make `save_account` non-`pub` and use this everywhere?

Reply:
Only place where it wasn't obvious was in exec.rs `apply_delta()`. It can of course be done but it didn't seem obviously simpler to me to refactor that way so I left it for now, to avoid even more changes in this PR - could open a new small PR if you think the refactor's worth it